core/src/main/scala/org/json4s/ParserUtil.scala

```diff
@@ -34,7 +34,12 @@ object ParserUtil {
         case '\r' => appender.append("\\r")
         case '\t' => appender.append("\\t")
         case c =>
-          if (!AsciiEncoder.canEncode(c))
+          val shouldEscape = if (formats.alwaysEscapeUnicode) {
+            !AsciiEncoder.canEncode(c)
```

Comment:
I just noticed this. Pretty sure this is actually still broken if you use the `alwaysEscapeUnicode` option.

`\u0009` is the tab character. It's an ASCII character so it can definitely be encoded. However, it's in the "must escape" range of characters for JSON.

I think the code should be something more like:

```scala
if((c >= '\u0000' && c <= '\u001f') || (c >= '\u0080' && c < '\u00a0') || (c >= '\u2000' && c < '\u2100') || (formats.alwaysEscapeUnicode && !AsciiEncoder.canEncode(c)))
  appender.append("\\u%04x".format(c: Int))
else appender.append(c.toString)
```